src/subsys/ngx_subsys_lua_util.c.tt2

```diff
@@ -4488,4 +4507,32 @@ ngx_[% subsys %]_lua_cleanup_free([% req_type %] *r,
 }
 
 
+#if (NGX_[% SUBSYS %]_LUA_HAVE_SA_RESTART)
+void
+ngx_[% subsys %]_lua_set_sa_restart(ngx_log_t *log)
+{
+    int                    *signo;
+    int                     sigs[] = NGX_[% SUBSYS %]_LUA_SA_RESTART_SIGS;
+    struct sigaction        act;
+
+    for (signo = sigs; *signo != 0; signo++) {
+        ngx_log_debug1(NGX_LOG_DEBUG_[% SUBSYS %], log, 0,
+                       "setting SA_RESTART for signal %d", *signo);
+
+        if (sigaction(*signo, NULL, &act) != 0) {
+            ngx_log_error(NGX_LOG_WARN, log, 0, "failed to get sigaction for "
+                          "signal %d (%s)", *signo, strerror(errno));
+        }
+
+        act.sa_flags |= SA_RESTART;
+
+        if (sigaction(*signo, &act, NULL) != 0) {
+            ngx_log_error(NGX_LOG_WARN, log, 0, "failed to set sigaction for "
+                          "signal %d (%s)", *signo, strerror(errno));
```

Comment:
This looks wrong. Why not replace the `0` errno argument directly with `ngx_errno`? Please address other similar places as well.

Reply:
Correct, updated!